Reusable post component

rohan2734 opened the pull request:
* this is a PR for issue #61 .
* currently there are comments, I will remove them as soon as the implementation is over.
* it is working fine for Pots_Details
* but it is not working for Pots on Home Page
* please I need a help at Posts on Home Page, please check where it is going wrong
* the share button and report are working on post_details, but not working on post
* console.log is working but the state is not updating in the posts of the home page.
* please give me good points, it took a lot of time.

rohan2734 commented:
@mann27  @ExpressHermes  please look at this PR.

ExpressHermes commented:
@rohan2734 there are conflicts in the branch. resolve them

rohan2734 commented:
> @rohan2734 there are conflicts in the branch. resolve them

* those were the commits which I have made, It is showing them as "conflicts".
* and please check the issue, the share button and the report button is working on post_details
* but not working on posts,
*actually the funciton call is correct, but the state is not being updated.
* please check,
*I will remove the comments then.

mann27 commented on debug.log:
```diff
@@ -0,0 +1 @@
+[1019/183503.273:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
remove this file.

rohan2734 replied:
ok, i think it was by mistake.

mann27 commented:
@rohan2734 The reason why the report and share are not working is there is no backend API to support that rn. They are yet to be implemented.
Report: The report-count of the post should increase after a report by the user. If that count surpasses a certain limit the post gets deleted automatically. ( this is the API that we have to create ). RN, it only shows reported by boolean state change just for showing the feature.
Share: this will copy the post link and then you can share the link will some other person no support of directly sharing this to some other platforms is there.

mann27 commented:
remove the comments and align it with the main branch.

> but it is not working for Pots on Home Page

I do not get your doubt, explain what is not working.

Once you do the changes, I will resolve the merge conflicts if any.

rohan2734 commented:
>I do not get your doubt, explain what is not working.

*previously, the share button and the report button were working on the posts on Home page,
*but now ,with that Reusable component , the share button and report button are not working on the posts of Home Page,
* but the share button and the report button are fine in posts of post_details page( when you click on a post)
* when i have checked that issue, the function call from post component to reusable component and viceversa is working
* but the copyText,reported , these two state values, are not working, theses are not getting updated via the reusable component
 in the posts of the home page.
* but these two states are getting updated on the post_details component(when we click on a post)
* the only problem is the copyText state, and reported state are not getting updated on the posts of the home page, via reusable component ,even though they are working in post_details component, via post_detials component.
* please check this.
* I will remove the comments once this is done.

rohan2734 commented:
@mann27 @ExpressHermes  did you guys check?

ExpressHermes commented:
> I will remove the comments once this is done.

@rohan2734 There are merge conflicts due to existing comments in one file. Please view the conflicts and see if you can remove the comments.

rohan2734 commented:
> @rohan2734 There are merge conflicts due to existing comments in one file. Please view the conflicts and see if you can remove the comments.

I removed the comments,  it is showing conflicts as i replaced the code with reusable component, not because of comments, now you can check them .
I have removed the comments.

mann27 commented:
give @rohan2734 75 points

mann27 commented:
The share was working on opening post cause it had those functions in that component itself and it was passed by props in index.js file.

Instead of writing the same functions twice, I included the CTA to reusable post component only so no need for inheritance.

The problem was with the hierarchy in components.
The following is how you call the actions of parent in child component.
```
import Parent from './parent.js';
class Child extends React.Component {
    constructor(props) {
        super(props);
        };

    click = () => {
        this.props.parentMethod();
    }

    render() {
          <div onClick={this.click}>Hello Child</div>
    }
}

class Parent extends React.Component {
    constructor(props) {
        super(props);
        };

    someMethod() {
        console.log('bar');
    }

    render() {
          <Child parentMethod={this.someMethod}>Hello Parent, {this.props.children}</Child>
    }
}
```

Hope this helps and look into the commit to see the changes.

rohan2734 commented:
>Instead of writing the same functions twice, I included the CTA to reusable post component only so no need for inheritance.

what is CTA?

>The share was working on opening post cause it had those functions in that component itself and it was passed by props in index.js file.

u mean post_details file?, i had taken the common functions and kept in reusable component, for post_detials/index.js it was working, but for  posts/index.js it was not working.

mann27 commented:
The functions in post_details were not moved so they were working and the posts functions were inherited through props and were not working.

CTA is short for Call To Action...i.e. a function
